lib/logstash/filters/geoip.rb

```diff
@@ -176,56 +184,40 @@ def filter(event)
 
     filter_matched(event)
   end # def filter
-  
-  def populate_geo_data(response, ip_address, geo_data_hash)
-    country = response.getCountry()
-    subdivision = response.getMostSpecificSubdivision()
-    city = response.getCity()
-    postal = response.getPostal()
-    location = response.getLocation()
-
-    # if location is empty, there is no point populating geo data
-    # and most likely all other fields are empty as well
-    if location.getLatitude().nil? && location.getLongitude().nil?
-      return
-    end
 
-    @fields.each do |field|
-      case field
-      when "city_name"
+  def populate_geo_data(response, ip_address, geo_data_hash)
+    case @dbtype
+      when "city"
+        country = response.getCountry()
+        subdivision = response.getMostSpecificSubdivision()
+        city = response.getCity()
+        postal = response.getPostal()
+        location = response.getLocation()
+
+        # if location is empty, there is no point populating geo data
+        # and most likely all other fields are empty as well
+        if location.getLatitude().nil? && location.getLongitude().nil?
+          return
+        end
         geo_data_hash["city_name"] = city.getName()
-      when "country_name"
         geo_data_hash["country_name"] = country.getName()
-      when "continent_code"
         geo_data_hash["continent_code"] = response.getContinent().getCode()
-      when "continent_name"
         geo_data_hash["continent_name"] = response.getContinent().getName()
-      when "country_code2"
         geo_data_hash["country_code2"] = country.getIsoCode()
-      when "country_code3"
         geo_data_hash["country_code3"] = country.getIsoCode()
-      when "ip"
         geo_data_hash["ip"] = ip_address.getHostAddress()
-      when "postal_code"
         geo_data_hash["postal_code"] = postal.getCode()
-      when "dma_code"
         geo_data_hash["dma_code"] = location.getMetroCode()
-      when "region_name"
         geo_data_hash["region_name"] = subdivision.getName()
-      when "region_code"
         geo_data_hash["region_code"] = subdivision.getIsoCode()
-      when "timezone"
         geo_data_hash["timezone"] = location.getTimeZone()
-      when "location"
         geo_data_hash["location"] = [ location.getLongitude(), location.getLatitude() ]
-      when "latitude"
         geo_data_hash["latitude"] = location.getLatitude()
-      when "longitude"
         geo_data_hash["longitude"] = location.getLongitude()
       else
-        raise Exception.new("[#{field}] is not a supported field option.")
-      end
+        geo_data_hash.merge!(JSON.parse(response.toJson()))
```

Comment:
Unfortunately, I don't know when I will be able to do that. If someone wants to continue from here, fine, if not, I will try to find a time for it in my next sprint.